src/net_processing.cpp

```diff
@@ -224,9 +224,23 @@ struct Peer {
 
     /** A vector of addresses to send to the peer, limited to MAX_ADDR_TO_SEND. */
     std::vector<CAddress> m_addrs_to_send;
-    /** Probabilistic filter of addresses that this peer already knows.
-     *  Used to avoid relaying addresses to this peer more than once. */
-    const std::unique_ptr<CRollingBloomFilter> m_addr_known;
+    /** Bloom filter to track recent addr messages relayed with this peer.
+     *
+     *  We initialize this filter for outbound peers (other than
+     *  block-relay-only connections) or when an inbound peer sends us an
+     *  address related message (ADDR, ADDRV2, GETADDR).
+     *
+     *  We use the presence of this filter to decide whether a peer is eligible
+     *  for trickle relay of addr messages. This avoids relaying to peers that
+     *  are unlikely to forward them, effectively blackholing self
+     *  announcements. Reasons peers might not support addr relay on the link
+     *  include that they connected to us as a block-relay-only peer or they
+     *  are a light client.
```

Comment:
This is no longer true. We use `addr_relay_enabled` to decide whether a peer is eligible for addr relay.

Reply:
ah good point. updated the comments on both `m_addr_known` and `m_addr_relay_enabled.